api/v1alpha1/nonadminbackupstoragelocation_types.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NonAdminBackupStorageLocationPhase is a simple one high-level summary of the lifecycle of an NonAdminBackupStorageLocation.
+// +kubebuilder:validation:Enum=New;Available;Unavailable;Created;Deleting
+type NonAdminBackupStorageLocationPhase string
+
+// NonAdminBackupStorageLocationPhase constants similar to velerov1.BackupStorageLocationPhase
+const (
+	NaBSLPhaseNew         NonAdminBackupStorageLocationPhase = "New"
+	NaBSLPhaseAvailable   NonAdminBackupStorageLocationPhase = "Available"
```

Comment:
is this phase used?

Reply:
Please see my comment below, I will adjust those phases to the:
https://github.com/migtools/oadp-non-admin/blob/25e40d6abd8c49c43a1a89a9f49d6d489d07707e/api/v1alpha1/nonadmin_types.go#L23-L32

The initial idea was to match the BSL but the BSL satus is fully reflected in the Status anyway so it makes sense to align with other NAC objects